File: nomad-samples/python/movies/create_movie/movies/search_movies.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def search_movies(AUTH_TOKEN, FILTERS, SEARCH_RESULT_FIELDS, SORT_FIELD, SORT_TYPE):
    # Create header for the request
    if not AUTH_TOKEN:
        raise Exception("Authorization token not found")
  
    API_URL = f"{PORTAL_API_URL}/portal/search"
    
    HEADERS = {
      "Content-Type": "application/json",
      "Authorization": "Bearer " + AUTH_TOKEN
    }

    DEFAULT_FILTERS = [
        {
            "fieldName": "contentDefinitionId",
            "operator": "Equals",
            "values": MOVIE_CONTENT_DEFINITION_ID,
        },
        {
            "fieldName": "languageId",
            "operator": "Equals",
            "values": US_ENGLISH_LANGUAGE_LOOKUP_ID,
        }
    ]

    for FILTER in FILTERS:
        DEFAULT_FILTERS.append(FILTER)
    

    # Build the payload body
    BODY = {
        "filters": DEFAULT_FILTERS,
        "searchResultFields": SEARCH_RESULT_FIELDS
    }
    

    if (SORT_FIELD != ""):
        BODY["sortFields"] = [
            {
                "fieldName": SORT_FIELD,
                "sortType": SORT_TYPE
            }
        ]
    

    # Send POST request
    RESPONSE = requests.post(API_URL, headers=HEADERS, data=json.dumps(BODY))

    # Check for success
    if (RESPONSE.ok):
        # Get the response
        MOVIES = RESPONSE.json()

        # Return the movies
        return MOVIES
    

    api_exception_handler(RESPONSE, "Getting movies failed")

File: nomad-samples/common/js/movies/create-movie/movie/search-movies.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function searchMovies(AUTH_TOKEN, PAGE_OFFSET, PAGE_SIZE, SEARCH_QUERY, FILTERS, SEARCH_RESULT_FIELDS, SORT_FIELD, SORT_TYPE, IS_ADMIN) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    const DEFAULT_FILTERS = [
        {
            fieldName: "contentDefinitionId",
            operator: "Equals",
            values: prjConstants.MOVIE_CONTENT_DEFINITION_ID,
        },
        {
            fieldName: "languageId",
            operator: "Equals",
            values: prjConstants.US_ENGLISH_LANGUAGE_LOOKUP_ID,
        },
    ]

    for(let filter_idx = 0; filter_idx < FILTERS.length; ++filter_idx)
    {
        DEFAULT_FILTERS.push(FILTERS[filter_idx]);
    }

    // Build the payload body
    const BODY = {
        filters: DEFAULT_FILTERS,
        SearchResultFields: SEARCH_RESULT_FIELDS
    };

    if (PAGE_OFFSET !== ""){BODY["pageOffset"] == PAGE_OFFSET}    
    if (PAGE_SIZE !== ""){BODY["pageSize"] == PAGE_SIZE}
    if (SEARCH_QUERY !== ""){BODY["searchQuery"] == SEARCH_QUERY}

    if (SORT_FIELD != "")
    {
        BODY.sortFields = [
            {
                fieldName: SORT_FIELD,
                sortType: SORT_TYPE
            }
        ]
    }

    // Send POST request
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/portal/search`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY),
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the response
        const MOVIES = await RESPONSE.json();

        // Return the movies
        return MOVIES;
    }

    apiExceptionHandler(RESPONSE, "Getting movies failed");
}
